### PR TITLE
 Updating Typescript Wrapper to 0.1.2-alpha-2 (with API key)

### PR DESCRIPTION
**Pull request by GDPWinnerPranata**
## Before testing
[Setup your own ZkPass Server](https://github.com/GDP-ADMIN/zkPass/blob/dev/ts-sdk-postinstall/README.md#initial-build-setup-prerequisites)

## How to test

1. Clone repository

### Client Setup
1. Change directory to Client
2. Install dependencies
3. Configure environment variable
4. Run Client

```bash
cd typescript/node-js/client
npm install
cp .env.example .env # and configure everything
```

```bash
nano .env

API_KEY=api_1
API_SECRET=secret_api_1
ZKPASS_SERVICE_URL="http://localhost:10888/proof"
```
```bash
npm run dev
```

### Issuer-Verifier Setup
1. Change directory to Issuer-Verifier
2. Install dependencies
3. Configure environment variable
4. Run Client

```bash
cd typescript/node-js/issuer-verifier
npm install
cp .env.example .env # and configure everything
```

```bash
nano .env

API_KEY=api_1
API_SECRET=secret_api_1
ZKPASS_SERVICE_URL="http://localhost:10888/proof"
```
```bash
npm run dev
```

### Testing UI
1. Open `http://localhost:3000`
2. Follow the instructions end-to-end

### CODE REVIEW

**Review by zulamdat — APPROVED** [review]

File: typescript/node-js/cli/src/Verifier.ts [hunk 1]
```diff
@@ -21,45 +21,70 @@ import {
   DataVerificationRequest,
   ZkPassClient,
   PublicKey,
+  MetadataValidatorResult,
+  PublicKeyWrapped,
+  KeysetEndpointWrapped,
+  VerifyZkpassProofResult,
 } from "@didpass/zkpass-client-ts";
-import { ZkPassProofMetadataValidator } from "@didpass/zkpass-client-ts/lib/interfaces";
+import { ZkPassProofMetadataValidator } from "@didpass/zkpass-client-ts";
```
Comment (LawrencePatrickSianto) [inline typescript/node-js/cli/src/Verifier.ts:29]: dijadikan satu dengan import _zkpass-client-ts_ yg atasnya

File: typescript/node-js/cli/src/utils/constants.ts [hunk 2]
```diff
@@ -17,9 +17,13 @@
...
 
+import { ZkPassApiKey } from "@didpass/zkpass-client-ts";
+import { config } from "dotenv";
+
+config();
+
 // Holder constants
-export const ZKPASS_SERVICE_URL: string =
-  "https://playground-zkpass.ssi.id/proof";
+export const ZKPASS_SERVICE_URL: string = "https://staging-zkpass.ssi.id/proof";
```
Comment (LawrencePatrickSianto) [inline typescript/node-js/cli/src/utils/constants.ts:26]: apa ngga mending URL ini dimasukkan ke .env?

**Comment by GDPWinnerPranata** [comment]
#48

File: typescript/node-js/cli/src/utils/constants.ts (hunk 2 above)
Comment (GDPWinnerPranata) [reply typescript/node-js/cli/src/utils/constants.ts:26]: krn CLI purposenya demo, semuanya kita masukin ke constant aja, termasuk private key & API key

**Review by LawrencePatrickSianto — APPROVED** [review]

**Review by zulamdat — APPROVED** [review]

**Pull request merged**